src/helics/cpp98/MessageFederate.hpp

```diff
@@ -110,5 +110,21 @@ class MessageFederate: public virtual Federate {
   private:
     std::vector<helics_endpoint> local_endpoints;
 };
+//this code needs the definition of federate before it can de defined
+inline Message::Message(const Federate& fed):
```

Comment:
Would it make sense to also have a version that takes a `Federate*` so users can pass a `fed` pointer instead of `*fed`?

Reply:
I thought about this, but Federate and Endpoint are basically wrappers around a void *,  so for the tests I wrap that in a shared_ptr so an overload that took a pointer wouldn't really help anything.  And in general Value Classes for these objects are preferred over pointers.  So I would lean towards not having a pointer overload.